File: LCX/ValidEndPoints/dsl/examples/authenticated_trading/authenticated_trading.ts
import crypto from 'crypto';
import fetch from 'node-fetch';


async function get_account_info(): Promise<any> {
  // GET /api/account
  const url = 'https://exchange-api.lcx.com/api/account';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/account';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_balances(): Promise<any> {
  // GET /api/balances
  const url = 'https://exchange-api.lcx.com/api/balances';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/balances';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_asset_balance(asset): Promise<any> {
  // GET /api/balance
  const url = 'https://exchange-api.lcx.com/api/balance';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/balance';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_trading_fees(): Promise<any> {
  // GET /api/fees
  const url = 'https://exchange-api.lcx.com/api/fees';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/fees';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_open_orders(offset, pair): Promise<any> {
  // GET /api/open
  const url = 'https://exchange-api.lcx.com/api/open';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/open';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_order_details(orderId): Promise<any> {
  // GET /api/order
  const url = 'https://exchange-api.lcx.com/api/order';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/order';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_order_history(offset, pair, fromDate, toDate, side, orderStatus, orderType): Promise<any> {
  // GET /api/orderHistory
  const url = 'https://exchange-api.lcx.com/api/orderHistory';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/orderHistory';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_user_trade_history(offset, pair, fromDate, toDate): Promise<any> {
  // GET /api/uHistory
  const url = 'https://exchange-api.lcx.com/api/uHistory';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/uHistory';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();
    return data;

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# POST method not supported for typescript
# PUT method not supported for typescript
# DELETE method not supported for typescript
# DELETE method not supported for typescript
async function check_balance_workflow(): Promise<any> {
  // GET /api/balances
  const url = 'https://exchange-api.lcx.com/api/balances';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/balances';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for typescript
async function place_order_workflow(BTC/USDC): Promise<any> {
  // GET /api/ticker
  const url = 'https://exchange-api.lcx.com/api/ticker';
  const headers = {'Content-Type': 'application/json'};

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for typescript
async function get_order_info(order_response): Promise<any> {
  // GET /api/order
  const url = 'https://exchange-api.lcx.com/api/order';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/order';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# None method not supported for typescript
async function manage_open_orders_workflow(0): Promise<any> {
  // GET /api/orderHistory
  const url = 'https://exchange-api.lcx.com/api/orderHistory';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/orderHistory';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# Workflow: ManageOpenOrdersWorkflow (sequence)

async function get_trade_history(0): Promise<any> {
  // GET /api/uHistory
  const url = 'https://exchange-api.lcx.com/api/uHistory';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/uHistory';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# None method not supported for typescript
# PUT method not supported for typescript
# DELETE method not supported for typescript
# None method not supported for typescript
async function account_monitoring_workflow(): Promise<any> {
  // GET /api/balances
  const url = 'https://exchange-api.lcx.com/api/balances';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/balances';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# Workflow: AccountMonitoringWorkflow (sequence)

async function get_open_orders(0): Promise<any> {
  // GET /api/open
  const url = 'https://exchange-api.lcx.com/api/open';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/open';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

async function get_fees(): Promise<any> {
  // GET /api/fees
  const url = 'https://exchange-api.lcx.com/api/fees';
  const headers = {'Content-Type': 'application/json'};
  const apiKey = process.env.LCX_API_KEY || '';
  const apiSecret = process.env.LCX_API_SECRET || '';
  const timestamp = Date.now().toString();
  const message = timestamp + 'GET' + '/api/fees';
  const signature = crypto.createHmac('sha256', apiSecret).update(message).digest('base64');
  headers['x-access-key'] = apiKey;
  headers['x-access-sign'] = signature;
  headers['x-access-timestamp'] = timestamp;

  try {
    const response = await fetch(url, {
      method: 'GET',
      headers: headers
    });
    const data = await response.json();

  } catch (error) {
    console.error('Error:', error);
    throw error;
  }
}

# None method not supported for typescript

File: LCX/ValidEndPoints/dsl/examples/authenticated_trading/authenticated_trading.go
package main

import (
    "fmt"
    "io"
    "net/http"
    "json"
    "os"
    "crypto/hmac"
    "crypto/sha256"
    "encoding/base64"
    "time"
)


func get_account_info() (map[string]interface{}, error) {
    // GET /api/account
    url := "https://exchange-api.lcx.com/api/account"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/account"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_balances() (map[string]interface{}, error) {
    // GET /api/balances
    url := "https://exchange-api.lcx.com/api/balances"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/balances"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_asset_balance(asset) (map[string]interface{}, error) {
    // GET /api/balance
    url := "https://exchange-api.lcx.com/api/balance"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/balance"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_trading_fees() (map[string]interface{}, error) {
    // GET /api/fees
    url := "https://exchange-api.lcx.com/api/fees"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/fees"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_open_orders(offset, pair) (map[string]interface{}, error) {
    // GET /api/open
    url := "https://exchange-api.lcx.com/api/open"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/open"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_order_details(orderId) (map[string]interface{}, error) {
    // GET /api/order
    url := "https://exchange-api.lcx.com/api/order"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/order"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_order_history(offset, pair, fromDate, toDate, side, orderStatus, orderType) (map[string]interface{}, error) {
    // GET /api/orderHistory
    url := "https://exchange-api.lcx.com/api/orderHistory"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/orderHistory"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_user_trade_history(offset, pair, fromDate, toDate) (map[string]interface{}, error) {
    // GET /api/uHistory
    url := "https://exchange-api.lcx.com/api/uHistory"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/uHistory"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# POST method not supported for go
# PUT method not supported for go
# DELETE method not supported for go
# DELETE method not supported for go
func check_balance_workflow() (map[string]interface{}, error) {
    // GET /api/balances
    url := "https://exchange-api.lcx.com/api/balances"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/balances"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# Workflow: CheckBalanceWorkflow (sequence)

# None method not supported for go
func place_order_workflow(BTC/USDC) (map[string]interface{}, error) {
    // GET /api/ticker
    url := "https://exchange-api.lcx.com/api/ticker"
    headers := map[string]string{
        "Content-Type": "application/json",
    }

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# Workflow: PlaceOrderWorkflow (sequence)

# POST method not supported for go
func get_order_info(order_response) (map[string]interface{}, error) {
    // GET /api/order
    url := "https://exchange-api.lcx.com/api/order"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/order"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# None method not supported for go
func manage_open_orders_workflow(0) (map[string]interface{}, error) {
    // GET /api/orderHistory
    url := "https://exchange-api.lcx.com/api/orderHistory"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/orderHistory"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# Workflow: ManageOpenOrdersWorkflow (sequence)

func get_trade_history(0) (map[string]interface{}, error) {
    // GET /api/uHistory
    url := "https://exchange-api.lcx.com/api/uHistory"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/uHistory"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# None method not supported for go
# PUT method not supported for go
# DELETE method not supported for go
# None method not supported for go
func account_monitoring_workflow() (map[string]interface{}, error) {
    // GET /api/balances
    url := "https://exchange-api.lcx.com/api/balances"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/balances"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# Workflow: AccountMonitoringWorkflow (sequence)

func get_open_orders(0) (map[string]interface{}, error) {
    // GET /api/open
    url := "https://exchange-api.lcx.com/api/open"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/open"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

func get_fees() (map[string]interface{}, error) {
    // GET /api/fees
    url := "https://exchange-api.lcx.com/api/fees"
    headers := map[string]string{
        "Content-Type": "application/json",
    }
    // HMAC-SHA256 Authentication
    apiKey := os.Getenv("LCX_API_KEY")
    apiSecret := os.Getenv("LCX_API_SECRET")
    timestamp := strconv.FormatInt(time.Now().UnixMilli(), 10)
    message := timestamp + "GET" + "/api/fees"
    h := hmac.New(sha256.New, []byte(apiSecret))
    h.Write([]byte(message))
    signature := base64.StdEncoding.EncodeToString(h.Sum(nil))
    headers["x-access-key"] = apiKey
    headers["x-access-sign"] = signature
    headers["x-access-timestamp"] = timestamp

    resp, err := http.Get(url)
    if err != nil {
        return nil, err
    }
    defer resp.Body.Close()

    body, _ := io.ReadAll(resp.Body)
    var result map[string]interface{}
    json.Unmarshal(body, &result)

}

# None method not supported for go
